src/components/PublisherView.js

```diff
@@ -15,37 +15,50 @@ const RNOpenTokPublisherView = requireNativeComponent(
 const publishListeners = ['onPublishStart', 'onPublishStop', 'onPublishError'];
 const NOOP = () => {};
 
-export default class PublisherView extends React.Component<PublisherViewProps> {
+export default class PublisherView extends React.Component<
+  PublisherViewProps,
+  { camera: number }
+> {
   static defaultProps = {
     onPublishStart: NOOP,
     onPublishStop: NOOP,
     onPublishError: NOOP,
     sessionId: '',
+    audioDisabled: false,
+    videoDisabled: false,
+  };
+
+  state = {
+    camera: 0,
   };
 
   componentWillMount() {
-    publishListeners.forEach(listener => this.addListener(listener));
+    publishListeners.forEach(listener => this._addListener(listener));
   }
 
   componentWillUnmount() {
-    publishListeners.forEach(listener => this.removeListener(listener));
+    publishListeners.forEach(listener => this._removeListener(listener));
   }
 
-  addListener = (name: string) => {
+  _addListener = (name: string) => {
     if (!this.props.listeners[name]) {
       this.props.listeners[name] = NativeEventEmitter.addListener(name, e =>
         this.props[name](e)
       );
     }
   };
 
-  removeListener = (name: string) => {
+  _removeListener = (name: string) => {
     if (this.props.listeners[name]) {
       this.props.listeners[name].remove();
       delete this.props.listeners[name];
     }
   };
 
+  switchCamera = () => {
+    this.setState(prevState => ({ camera: prevState.camera + 1 }));
```

Comment:
I don't get it this part. It's not a toggle implementation. Have I missed something?

Reply:
If it just about to trigger the props in native side (Andorid) we could do it also with string to make it more clear, don't you think? 💃